### PR TITLE
OAP-17: PostgreSQL integration into API with pg-promise, data function to read from DB, dotenv to read DB credentials from environment variables

### PR DESCRIPTION
**Pull request by j-sofia**
added pg-promise package to interface with PostgreSQL
added data functions to read from the suggestions table
changed api to port 3001
updated README.md

### CODE REVIEW

File: api/db/data.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const validate = require("../validate");
+const { ParameterizedQuery: PQ } = require("pg-promise");
+
+const db = require("./connection.js");
+
+async function querySuggestions(id) {
+  await validate.checkHandle(id);
+
+  // Create a parameterized query with the id
+  const query = new PQ({ text: "SELECT * FROM oapen_suggestions.suggestions WHERE item_id = $1", values: [id] });
```
Comment (justinoboyle) [inline api/db/data.js:10]: Might want to break this out into a library function

File: api/package.json [hunk 2]
```diff
@@ -4,11 +4,14 @@
   "description": "An API which allows you to search for books with similar content from the oapen.org library.",
   "main": "app.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "seed": "node ./db/seed.js",
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (justinoboyle) [inline api/package.json:8]: can remove test script for now

File: api/db/data.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+const validate = require("../validate");
```
Comment (justinoboyle) [inline api/db/data.js:1]: is `validate.js` checked in?

**Comment by justinoboyle** [comment]
rest looking good!

File: api/db/data.js (hunk 1 above)
Comment (j-sofia) [reply api/db/data.js:10]: The query? How?

Comment (justinoboyle) [reply api/db/data.js:10]: absolutely not a big deal at all, so low prio, but I usually keep routes and library/helper functions seperate, could make a `getSuggestionsByBookID(id)` that could be used in a few places. again not a big deal at all, a small nitpick haha

Comment (eshellman) [reply api/db/data.js:10]: I'm not familiar with JS libraries - does ParameterizedQuery handle parameter sanitization?

Comment (j-sofia) [reply api/db/data.js:10]: It should handle SQL injection https://github.com/brianc/node-postgres/wiki/FAQ#8-does-node-postgres-handle-sql-injection
And we will code out the validate.checkHandle(id) function to further validate the input

File: api/db/data.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  // Create a parameterized query with the id
+  const query = new PQ({ text: "SELECT * FROM oapen_suggestions.suggestions WHERE item_id = $1", values: [id] });
+
+  // Run query
+  return db.one(query).catch((error) => {
+    return { error: { name: error.name, message: error.message } };
+  });
+}
+
+let exportedMethods = {
```
Comment (celinanperalta) [inline api/db/data.js:18]: Style nit: This could be exported as one function

File: api/db/seed.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+require("dotenv").config({ path: "./db/database.env" });
+const db = require("./connection.js");
+
+//seed can be run from "make setup-env" from oapen-engine/
```
Comment (celinanperalta) [inline api/db/seed.js:4]: Move to README

File: api/routes.js [hunk 6]
```diff
@@ -2,14 +2,30 @@ const express = require("express");
...
-    let responseData = { error: "Not implemented" };
+    var handle = req.params.handle;
+
+    //Validate the parameter
+    await validate.checkHandle(handle);
+
+    //Get the data
+    let responseData = await data.getSuggestions(handle);
+
+    if (responseData.error && responseData.error.name === "QueryResultError") {
```
Comment (celinanperalta) [inline api/routes.js:18]: Use `pgp.errors.QueryResultError.name`

**Review by celinanperalta — COMMENTED** [review]
Good, added some nitpicks and I suggest removing comments that are explained by the code itself.

**Comment by justinoboyle** [comment]
Need to merge master & resolve conflicts -- most can be ignored but make sure to take `process.env.PORT` into account for launch scripts.

**Review by peterrauscher — APPROVED** [review]

**Pull request merged**